### ISSUE TITLE
Components like assets

### ISSUE THREAD

**Issue by rjs5327** [snipe/snipe-it#6207]
**Server (please complete the following information):**
Web
 - Snipe-IT Version 
v4.6.4
 - OS: [e.g. Ubuntu, CentOS]
 - Web Server: [e.g. Apache, IIS]
 - PHP Version 

**Is your feature request related to a problem? Please describe.**
I would like to see the be able to be a component created. Say 3.5" 4 TB Hard Drive Western Digital Manufacture. Then i can say create new assign the purchase date and serial. Right now i can do this but i have to creat new component every time if i want to put multiple purchases and or serial numbers in.

**Describe the solution you'd like**
I would like to see a sub table in the components creted that allows for items of the component. So you have a master record, having the static info. Like manufacture and model. Then a sub table tied to the component id that has purchase price and serial. This way one could create a item like hard drives and then pick them to a NAS or computer. Reason to do this is to 1. Show value of asset increased which would need modified and 2. Track life of components. Does my drive last 2 years 10 years or so on. 

**Describe alternatives you've considered**
Basically i am suggesting making components like assets except for no ID this way i can buy 10 of something indicate there serials and pick it to a asset. 

**Additional context**
Components when added to a asset should change a value called asset value. This is the appreciated cost or value of the item.

**Comment by ccomtest** [comment]
I do something like this with creating a component and don't put a serial here.
I create a custom field with serial and connect to an asset.
Then put this to "archive" type and cone it from there to an assignable asset.
You just need to type serial in over custom field.
Small problem left (i opened an request about this) you have to put every component to an asset one by one. If you have a lot of them (bundles like me) you will need a lot of time for searching and connecting components with assets. If you could assign much components to many assets, this way would work fine i think.

**Comment by snipe** [comment]
Why don't you just create them as assets?

**Comment by jflatz** [comment]
I too would like to see the ability to treat components more like assets with the ability to have individual purchase dates, purchase price, serial numbers, etc. Also, I believe issue #4421 is related.

**Comment by jflatz** [comment]
> Why don't you just create them as assets?

To be fair, with that logic, why does the app even have components than?

**Comment by spotter-mmf** [comment]
I'd definitely like to see this added as a feature! I'd also like to be able to track status of a component.

Example: I have a 4 bay NAS. I'd like to be able to check out hard drives to fill the bays of that NAS. When a drive fails, I'd like to be able to check the drive in, change the status to pending and mark it as out for repair. 

These are all actions that we can perform against assets but not components. If we should be entering these items as assets, what is the point of having a components type?

**Comment by dbinoj** [comment]
I would also like to have this feature / functionality in components. Feels like a valid use case

**Comment by kunamashina** [comment]
This is very required. Right now there's no way to enter components with their serials and track their availability. Only thing I can do is add each component with the serial as assets and add them to other assets.

My case was:
I had 5 Kingston SSDs of same model and same size but different serials. So when I enter them as 5 pieces I don't know which SSD is in which laptop???

Instead I had to add them as assets and assign them to other assets (laptops) which makes components obsolete in the end and useless...

**Comment by uFojun** [comment]
I completely agree with the last comment
Asset-controlled computers, servers, monitors or end devices. And I did not want to add case disks or memory to this primary list. But at the same time, no one canceled the serial numbers for these components.
Thank you snipe-it team. You are cool!

**Comment by mylittleserver** [comment]
That would be a great feature.

**Comment by KyleTryon** [comment]
I just discovered this tool today and the first thing I looked for was this. Very much in line with what kunamashina and spotter-mmf 


Looking at this from a computer repair perspective, you want to track your parts the same as any other asset. All of which have costs associated with them. So if you have an inventory of hard drives (which cost $X) I'd like to remove one of those drives from the available inventory and include is as a component of a computer in stock, which would then add on to the total cost of that computer.

**Comment by cadlebe** [comment]
Can we see this issue as reopened. Being able to track internal components would be a boon from a maintenance perspective by giving us a history on specific components. Like others have said I don't want to track each circuit board, NIC, and HDD as an asset.

**Comment by giorgi-gagnidze** [comment]
Hello.


> This is very required. Right now there's no way to enter components with their serials and track their availability. Only thing I can do is add each component with the serial as assets and add them to other assets.
> 
> My case was:
> I had 5 Kingston SSDs of same model and same size but different serials. So when I enter them as 5 pieces I don't know which SSD is in which laptop???
> 
> Instead I had to add them as assets and assign them to other assets (laptops) which makes components obsolete in the end and useless...

This is problem for me also, I have laptops with default configuration but also some components added inside, e.g. SSD or RAM and I need to track manufacturer, serial number, price, etc. There will be good thing if components also have fields like other groups. 
Manufacturer, custom fields, serial, warranty.
and also I can't understand, how can I add e.g. SSD to laptop or PC?
Thank you.

**Comment by Oddar** [comment]
This is still a valid feature request.
Adding 20 SSDs as components, but being able to control each of those twenty disks as separate objects by serial number.
Only way to achieve this now is to add them either as assets (makes a messy asset-list), or adding the component twenty times.

What would be awesome: Have GUI just the way it is at the moment, but have the field “serial number” accept comma separated values. Then, when adding one of these components to an asset - you are asked to chose which serial number you want to check out to that asset. If no serial number is given, or only one serial number, the software should behave like it is now.

I guess what would make this even more intuitive, would be to have a check box when adding the components asking whether or not I want to control these components individually by serial number. If checked, it would require serial numbers for each components. 

For some, the most important thing is how many SSD are still available in inventory - and current functionality will suffice.
For others, they need to control components by individual - maybe for compliance either for infosec, or maybe for fiscal matters, or maybe just for having our own “statistics” with regards to lifetime of the components.

**Comment by dbinoj** [comment]
I think stalebot closed this issue prematurely. Looks like this feature, if implemented, would be used by many and would also not affect people who use it the same way it is right now. It would also help differentiate low cost consumables from high cost components which are not "assets" but help people track them individually since they are of high value.

@snipe Would it be appropriate to open this as a new issue or reopen this so that this feature request, write ups of ways people may use it and the demand for it is not forgotten? If something like this is already in your roadmap, these discussions may help shape them as well!

**Comment by dbinoj** [comment]
@snipe Sorry, just read #8473 as well as other similar open issues. Please ignore previous comment.

**Comment by aslowwriter** [comment]
would also love this feature, basically for the NAS use case already described above.

**Comment by elvinu** [comment]
I was googleing to find how people are dealing with this, and found this thread. I hope it will be implemented in some form in the future.🙏

**Comment by aramirez-capacity** [comment]
I just began to use Snipe-IT and found this feature missing. Are there any plans to implement it?

**Comment by snipe** [comment]
No. Just create them as assets, as I mentioned over and over in this thread.

**Comment by KyleTryon** [comment]
If it's been said enough times, perhaps it should be a feature.

On Aug 5, 2021 12:07 PM, snipe ***@***.***> wrote:

No. Just create them as assets, as I mentioned over and over in this thread.

**Comment by snipe** [comment]
Or you could just take my advice. 

We are not building that "feature" because it's a misfeature. If it should be treated like an asset, it's an asset.
